### ISSUE TITLE
RFE : Les deux compétences de wr et en particulier, l'analyse d'une route créée manuellement par l'utilisateur.

### ISSUE THREAD

**Issue by Gilletarom** [rgleason/weather_routing_pi#143]
WR is a plugin that had one skill: "Create routes to go as fast as possible from a starting point to an ending point."
BUT
Now, he theoretically has a second skill: "To analyze a route created manually by the user" and:
- on the one hand, declare if this route is viable, i.e. it can be traveled by the boat
- on the other hand, if it is viable, in how long can it be covered.

As things currently stand, the operation of this second skill is as follows:
- the route to be analyzed is designated using the "Analyze a route" item in the main menu of OpenCPN
  
![2023-09-08 22_11_44-OpenCPN 5 8 4-0](https://github.com/rgleason/weather_routing_pi/assets/6445543/6469d6d6-a372-4f87-a7c8-0f307709c5ae)

- WR creates two weather-routing positions corresponding respectively to the first waypoint of the route and to the last waypoint of the route.
- WR prepares a new "Search Configuration" of routes as if WR should search for the best possible route between these two weather-routing positions, but this is not what WR should do.

![2023-09-08 22_14_40-Météo-routing - WeatherRoutingConfiguration xml](https://github.com/rgleason/weather_routing_pi/assets/6445543/98446288-2875-4989-a168-b8ef5885572c)

- The user must complete the parameters of this search configuration, in particular by including the expected departure time
- The user must trigger the calculations for this "Search Configuration".
 
- If the route is not viable, the error(s) that are displayed resemble the errors encountered by WR to find the best route.
- If the route is viable, a meteo-trace is created to know the travel times of each segment and the times of passage on each waypoint of the route.

If the route is not viable, you must display the list of the impossibilities to meet:
- if the road crosses the earth, it must be clearly stated.
- if the road crosses a boundary, this must be stated clearly.
- if the boat cannot cover a segment, because the wind cannot be exploited (headwind for example) and this must be clearly stated.
- as for the "polar" error, I don't really understand what it means in this case. It must also probably be explained, we can expect it to be explained by the fact that the polar of the boat is of "poor quality".

If the route is viable, then:
- With the new export weather-routes button, it is not necessary to create a new route in the OpenCPN route manager. Simply complete the road properties table by incorporating the data calculated by WR.
- The creation of a meteo-trace of this route is only of interest if one wants to be able to compare, after having traveled the route actually, the meteo-trace with the real trace recorded by OpenCPN in the trace manager. A tool for displaying this comparison must be created.

**Comment by rgleason** [comment]
Yes this is certainly possible, but these features must be simple and effective.
Actually WR is quite flexible as it stands, using the configuration > constraints which allow consideration of wave ht and wind and comfort etc.

**Comment by Gilletarom** [comment]
Rick,

For information :

I had created a route with WR. It occurred to me, with this route, to 
have it tested by WR as if it were a route created manually by a user.

On this test, WR was not able to check the route.

I haven't repeated this type of test, but it intrigued me (and worried 
me, but this is the first time I'm talking about it).

Gilletarom

Le 08/09/2023 à 22:51, Rick Gleason a écrit :

**Comment by Gilletarom** [comment]
I wanted to test the comparison between two routes by comparing the property windows of each of these two routes.

BUT : 
Currently, OpenCPN does not allow two properties windows to be opened in parallel. You have to open one, then close it, then open the other.

It is not practical.

**Comment by rgleason** [comment]
Yes, only 1 properties window can be opened, and nothing else can be done until closed, it is a closed mode. This is normal. Nothing we can do.

**Comment by Gilletarom** [comment]
No, this way of approaching the problem which consists of saying: "It's like that and there's nothing we can do about it" is unacceptable. On the other hand, I can understand that the time available to resolve this is insufficient.

**Comment by rgleason** [comment]
Please explain in one simple sentence what you want to do, not how to do it though.

Then explain in one simple sentence how you think it might be accomplished.

**Comment by rgleason** [comment]
Gilletarom,

This has become more of a discussion, and is not a single issue.

> On this test, WR was not able to check the route.

Yes, that is correct. WR creates routes, it does not check them.

**Comment by rgleason** [comment]
@Gilletarom  Thinking has evolved since this dialog. Sebastien has suggested simply using Marks and Waypoints for weather_routing.  I like his idea but have no idea what it will take to impliment it. However this would probably enable users to adjust thooe location of the Marks, and do all the things that can be done to Marks, including change the icon, delete, move, etc.

I think that it much more likely that you will be able to use Weather_routing to check a route to see if it is possible under the predicted conditions, if we convert from using special wxroute marks to using standard OpenCPN marks.

However I do not know all that that entails.  Does this address  your issue above?
